packages/backend-common/src/service/lib/config.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigReader } from '@backstage/config';
+import { CorsOptions } from 'cors';
+
+export type BaseOptions = {
+  bindPort?: number;
+  bindHost?: string;
+};
+
+/**
+ * Reads some base options out of a config object.
+ *
+ * @param config The root of a backend config object
+ * @returns A base options object
+ *
+ * @example
+ * ```json
+ * {
+ *   baseUrl: "http://localhost:7000"
+ *   bindPort: 7000
+ *   bindHost: "0.0.0.0"
+ * }
+ * ```
+ */
+export function readBaseOptions(config: ConfigReader): BaseOptions {
+  return removeUnknown({
+    bindPort: config.getOptionalNumber('bindPort'),
+    bindHost: config.getOptionalString('bindHost'),
+  });
+}
+
+/**
+ * Attempts to read a CORS options object from the root of a config object.
+ *
+ * @param config The root of a backend config object
+ * @returns A CORS options object, or undefined if not specified
+ *
+ * @example
+ * ```json
+ * {
+ *   cors: {
+ *    origin: "http://localhost:3000",
+ *    credentials: true
+ *   }
+ * }
+ * ```
+ */
+export function readCorsOptions(config: ConfigReader): CorsOptions | undefined {
+  const cc = config.getOptionalConfig('cors');
+  if (!cc) {
+    return undefined;
+  }
+
+  return removeUnknown({
+    origin: getOptionalStringOrStrings(cc, 'origin'),
+    methods: getOptionalStringOrStrings(cc, 'methods'),
+    allowedHeaders: getOptionalStringOrStrings(cc, 'allowedHeaders'),
+    exposedHeaders: getOptionalStringOrStrings(cc, 'exposedHeaders'),
+    credentials: cc.getOptionalBoolean('credentials'),
+    maxAge: cc.getOptionalNumber('maxAge'),
+    preflightContinue: cc.getOptionalBoolean('preflightContinue'),
+    optionsSuccessStatus: cc.getOptionalNumber('optionsSuccessStatus'),
+  });
+}
+
+function getOptionalStringOrStrings(
```

Comment:
Just always require it to be an array instead imo

Reply:
Hmmm. But why :) anyway the cors config supplies either so we might just as well too. And some of these fields are like almost ALWAYS just one value, so it's syntactically annoying to make them lists. Can imagine the frustrated users "gah why won't it load, oh pff have to put that junk in there" :)

Reply:
I guess we could also just make `config.getStringArray` allow strings as well and just wrap it up in an array automatically?